node/src/main/scala/coop/rchain/node/NodeRuntime.scala

```diff
@@ -320,10 +320,12 @@ class NodeRuntime private[node] (
       blockMap,
       Metrics.eitherT(Monad[Task], metrics)
     )
-    _              <- blockStore.clear() // TODO: Replace with a proper casper init when it's available
-    oracle         = SafetyOracle.turanOracle[Effect](Monad[Effect])
-    runtime        = Runtime.create(storagePath, storageSize, storeType)
-    _              <- runtime.injectEmptyRegistryRoot[Effect]
+    _       <- blockStore.clear() // TODO: Replace with a proper casper init when it's available
+    oracle  = SafetyOracle.turanOracle[Effect](Monad[Effect])
+    runtime = Runtime.create(storagePath, storageSize, storeType)
+    _ = Runtime
+      .injectEmptyRegistryRoot[Task](runtime.space, runtime.replaySpace)
+      .unsafeRunSync(scheduler)
```

Comment:
This seems bad. If you can't use `Effect` directly anymore then you can still lift the `Task` into `Effect` using `toEffect` rather than using `unsafeRunSync`. E.g. https://github.com/rchain/rchain/blob/dev/node/src/main/scala/coop/rchain/node/NodeRuntime.scala#L383

Reply:
Good catch